packages/front-end/components/Settings/Billing/CloudUsage.tsx

```diff
@@ -0,0 +1,287 @@
+
+// Formatter for numbers
+const requestsFormatter = new Intl.NumberFormat("en-US", {
+  notation: "compact",
+  compactDisplay: "short",
+});
+
+function formatBytes(bytes: number) {
+  if (bytes === 0) return "0 Bytes";
+
+  const k = 1024;
+  const sizes = ["Bytes", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
+
+  const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+  return parseFloat((bytes / Math.pow(k, i)).toFixed(0)) + " " + sizes[i];
+}
+
+export default function CloudUsage() {
+  const [monthsAgo, setMonthsAgo] = useState(0);
+
+  const gb = useGrowthBook();
+
+  const router = useRouter();
+  const useDummyData = !isCloud() && !!router.query.dummy;
+
+  const enableUsage = useDummyData || gb.isOn("cdn-usage-data");
+
+  const { data, error } = useApi<{ cdnUsage: DailyUsage[] }>(
+    `/billing/usage?monthsAgo=${monthsAgo}`,
+    {
+      shouldRun: () => enableUsage && !useDummyData,
+    }
+  );
+
+  if (!enableUsage) return null;
+
+  if (error) {
+    return (
+      <Callout status="error">
+        Failed to get usage data: {error.message}
+      </Callout>
+    );
+  }
+
+  const usage = data?.cdnUsage || [];
+
+  // Use dummy data for testing
+  if (useDummyData) {
+    const startDate = new Date();
+    startDate.setUTCMonth(startDate.getUTCMonth() - monthsAgo);
+    startDate.setUTCDate(1);
+    startDate.setUTCHours(0, 0, 0, 0);
+
+    // Current day of month - all numbers after this will be 0
+    const CURRENT_DAY = monthsAgo > 0 ? 99 : 20;
+
+    // Generate dummy data for every day in the selected month
+    const now = new Date(startDate);
+    for (let i = 0; i < 32; i++) {
+      // Stop when we reach the next month
+      if (now.getUTCMonth() !== startDate.getUTCMonth()) break;
+      usage.push({
+        date: new Date(now).toISOString(),
+        requests: i >= CURRENT_DAY ? 0 : Math.floor(Math.random() * 10000000),
+        bandwidth:
+          i >= CURRENT_DAY ? 0 : Math.floor(Math.random() * 10000000000),
+      });
+      now.setUTCDate(now.getUTCDate() + 1);
+    }
+  }
+
+  const totalRequests = usage.reduce((sum, u) => sum + u.requests, 0);
+  const totalBandwidth = usage.reduce((sum, u) => sum + u.bandwidth, 0);
+
+  const monthOptions: { value: string; label: string }[] = [];
+  for (let i = 0; i < 12; i++) {
+    const date = new Date();
+    date.setUTCMonth(date.getUTCMonth() - i);
+
+    // Skip months before Feb 2025
+    if (date < new Date("2025-02-01")) continue;
+
+    const month = date.toLocaleString("default", {
+      month: "short",
+      timeZone: "UTC",
+    });
+    const year = date.getUTCFullYear();
+    monthOptions.push({
+      value: i + "",
+      label: `${month} ${year}`,
+    });
+  }
+  const maxMonthsAgo = monthOptions.length - 1;
+
+  return (
+    <Frame style={{ position: "relative" }}>
+      {!usage.length && <LoadingOverlay />}
+      <Flex gap="2" align="center" mb="4">
+        <h3 className="mr-4 mb-0">CDN Usage</h3>
```

Comment:
What are your thoughts on displaying the percentage of their available bandwidth they've used here to add context about these numbers.

For now we could just define the limits in code, but eventually we can pull the limits from the orgs subscription.